### PR TITLE
Wrapper function for private 1st and 2nd order allpass implementations

### PR DESCRIPTION
**Pull request by hoyer-a**
Wrapper function for allpass implementation #527 

I probably should add a test for the filter properties to ``test_filter.py``. Any ideas for a smart way of checking if the filter is doing what it's supposed to do?

Link to docs:
https://pyfar--571.org.readthedocs.build/en/571/modules/pyfar.dsp.filter.html#pyfar.dsp.filter.allpass

### CODE REVIEW

File: tests/test_filter.py [hunk 1]
```diff
@@ -127,6 +127,44 @@ def test_bessel(impulse):
...
+        assert f_obj.comment == (
+            f"{kind} order allpass-filter with cutoff frequency {fc} Hz.")
+
+        # Filter
+        x = pfilt.allpass(impulse, fc, order)
+        y = f_obj.process(impulse)
+        assert isinstance(x, Signal)
+        npt.assert_allclose(x.time, y.time)
+
+    # test ValueError
```
Comment (f-brinkmann) [inline tests/test_filter.py:150]: I would suggest to make a separate test function for the assertions below

File: tests/test_filter.py [hunk 2]
```diff
@@ -127,6 +127,44 @@ def test_bessel(impulse):
         x = pfilt.bessel(None, 2, 1000, 'lowpass', 'phase')
 
 
+def test_allpass(impulse):
+    # Uses third party code.
+    # We thus only test the functionality not the results
+
+    fc = 1000
+    orders = [1, 2]
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply tests/test_filter.py:135]: You can also prove more arguments in parametrize. For example you do 
`@pytest.parametrize('order, group_delay, frequency', ([1, group_delay_1, frequency_1], [2, group_delay_2, frequency_2]))` and use this to test if the group delay of the allpass is close to `group_delay` below `frequency`

[the remark this answers is not in the document]
Comment (hoyer-a) [reply tests/test_filter.py:135]: Thanks for the tip.

Now i'm testing the group delay $T_{gr}$ at ``fc`` for $T_{gr}(f_c) = T_{gr}(f_0) * \frac{1}{\sqrt{2}}$ according to the definition in Tietze/Schenk. 

And the difference between $T_{gr}(f_0)$ and $T_{gr}(f_c)$ is my tolerance for frequencies below ``fc``. Hope that makes any sense :D

File: pyfar/dsp/filter/audiofilter.py [hunk 3]
```diff
@@ -4,6 +4,134 @@
...
+    fs = signal.sampling_rate if sampling_rate is None else sampling_rate
+
+    if order == 1:
+        if coefficients is None:
+            coefficients = 0.6436
+        # get filter coefficients for first order allpass
+        _, _, b, a = iir.biquad_ap1st(frequency, fs, ai=coefficients)
+    elif order == 2:
+        if coefficients is None:
+            coefficients = [0.8832, 1.6278]
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:114]:
```suggestion
            coefficients = [1.6278, 0.8832]
```
it does not match the documentation above.

File: pyfar/dsp/filter/audiofilter.py [hunk 4]
```diff
@@ -4,6 +4,134 @@
...
+
+        -   For 2nd order allpass provide coefficients as list ``[bi, ai]``.\n
+            The default is ``bi = 1.6278``, ``ai = 0.8832``.
+
+        Defaults are chosen according to Tietze et al. (Fig. 12.66)
+        for maximum flat group delay.
+    sampling_rate : None, number
+        The sampling rate in Hz. Only required if signal is ``None``. The
+        default is ``None``.
+    Returns
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:50]:
```suggestion

    Returns
```

File: pyfar/dsp/filter/audiofilter.py [hunk 5]
```diff
@@ -4,6 +4,134 @@
...
+    -------
+    signal : Signal
+        The filtered signal. Only returned if ``sampling_rate = None``.
+    filter : FilterIIR
+        Filter object. Only returned if ``signal = None``.
+    References
```
Comment (ahms5) [inline pyfar/dsp/filter/audiofilter.py:56]:
```suggestion

    References
```

**Review by ahms5 — CHANGES_REQUESTED** [review]
Thanks for implementing! Just small comments

File: pyfar/dsp/filter/audiofilter.py (hunk 3 above)
Comment (hoyer-a) [reply pyfar/dsp/filter/audiofilter.py:115]: The coefficients in the docstring were swapped.

**Review by ahms5 — DISMISSED** [review]

**Review by f-brinkmann — DISMISSED** [review]

File: pyfar/dsp/filter/audiofilter.py [hunk 6]
```diff
@@ -4,6 +4,136 @@
...
+    frequency and are often used in analogue loudspeaker design.
+    The filter transfer function is based on Tietze et al. [#]_:
+
+    .. math:: A(s) = \\frac{1-\\frac{a_i}{\\omega_c} s+\\frac{b_i}
+                {\\omega_c^2} s^2}{1+\\frac{a_i}{\\omega_c} s
+                +\\frac{b_i}{\\omega_c^2} s^2},
+
+
+    where :math:`\\omega_c = 2 \\pi f_c` with the cut-off frequency :math:`f_c`
+    and :math:`s=\\sigma + \\mathrm{i} \\omega`.
```
Comment (mberz) [inline pyfar/dsp/filter/audiofilter.py:21]:
```suggestion
    .. math:: A(s) = \\frac{1-\\frac{a_i}{\\omega_c} s+\\frac{b_i}
                {\\omega_c^2} s^2}{1+\\frac{a_i}{\\omega_c} s
                +\\frac{b_i}{\\omega_c^2} s^2},


    where :math:`\\omega_c = 2 \\pi f_c` with the cut-off frequency :math:`f_c`
    and :math:`s=\\mathrm{i} \\omega`.
```
In our case we do not use the real part of the Laplace transform.

File: pyfar/dsp/filter/audiofilter.py [hunk 7]
```diff
@@ -4,6 +4,136 @@
...
+        raise ValueError('Coefficients must match the allpass order')
+
+    # sampling frequency in Hz
+    fs = signal.sampling_rate if sampling_rate is None else sampling_rate
+
+    if order == 1:
+        if coefficients is None:
+            coefficients = 0.6436
+        # get filter coefficients for first order allpass
+        _, _, b, a = iir.biquad_ap1st(frequency, fs, ai=coefficients)
```
Comment (mberz) [inline pyfar/dsp/filter/audiofilter.py:113]:
```suggestion
        b, a = iir.biquad_ap1st(frequency, fs, ai=coefficients)[2, 3]
```

File: pyfar/dsp/filter/audiofilter.py [hunk 8]
```diff
@@ -4,6 +4,136 @@
...
+    elif order == 2:
+        if coefficients is None:
+            coefficients = [0.8832, 1.6278]
+        # get filter coefficients for second order allpass
+        _, _, b, a = iir.biquad_ap2nd(frequency, fs, bi=coefficients[0],
+                                      ai=coefficients[1])
```
Comment (mberz) [inline pyfar/dsp/filter/audiofilter.py:119]:
```suggestion
        b, a = iir.biquad_ap2nd(
            frequency, fs, bi=coefficients[0], ai=coefficients[1])[2, 3]
```

**Review by mberz — CHANGES_REQUESTED** [review]
Couple of small things

**Review by f-brinkmann — APPROVED** [review]

**Pull request merged**